### ISSUE TITLE
Master React, Quick Code tutorial. Part 3

### ISSUE THREAD

**Issue by atherdon** [GroceriStar/react-intern-only-27#4]
https://medium.com/quick-code/lets-learn-react-chapter-3-components-components-components-3492f771d623

## Expected result

- [x] Read this tutorial chapter
- [x] follow code steps
- [x] move your code to this repository(at least one commit per day)
- [x] explain to me, what did you learn from it in a few sentences(not less than 5)
- [x] pass the test section
- [x] after reviewing your progress and closing this issue - you can move to the next chapter.

## Test
- [x] similar component for day checker is month checker. Create it.
- [x] btw, can you have a some sort of component between month and day - week, right?
with ability(method) - to grab only work days, and other method for weekends?
- [x] separate some of repeating functionality from that HTML code too...
- [x] I want to have one array - that will contain days and month together. so when you'll need to render a 'calendar' table - you'll use 2 forEach methods.
- [x] and i want to have all static data - like titles, image url, content - into one file, make it like array. sample: https://github.com/ChickenKyiv/recipe-search-react/tree/master/src/data
#1

**Comment by xFloki** [comment]
On the tutorial I saw how to organize better my components, how they should have a specific purpose so later on can be re-used. 
When returning multiple elements on the render of a component they must be wrapped or inside an array otherwise we'll get an error.

I have a question regarding of having all the static data into on file, should all the static text of my app go there as a good practice? When its full sentences I don't have a problem but when I'm using string interpolation I always struggle to know what's the better approach seems weird to create a constant for a fragment sentence.

PR Done, let me know if anything could be improved

**Comment by atherdon** [comment]
passing static text/labels into separate file is just a habit. in previous
company we tried to make websites templates more clear so this approach
helping, it also good thing, if you have a multy language system. So for me
- it's just an option to make code more elegant. :)


On 08-Jul-2018 at 02:30 PM, Alejandro Dietta wrote:


On the tutorial I saw how to organize better my components, how they should
have a specific purpose so later on can be re-used.
When returning multiple elements on the render of a component they must be
wrapped or inside an array otherwise we'll get an error.

I have a question regarding of having all the static data into on file,
should all the static text of my app go there as a good practice? When its
full sentences I don't have a problem but when I'm using string
interpolation I always struggle to know what's the better approach seems
weird to create a constant for a fragment sentence.

PR Done, let me know if anything could be improved

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-intern-only-27/issues/4#issuecomment-403291688>,
or mute the thread
